phoenix-core/src/it/java/org/apache/phoenix/end2end/AlterTableWithViewsIT.java

```diff
@@ -810,59 +827,65 @@ public void testDivergedViewsStayDiverged() throws Exception {
             PTable table = PhoenixRuntime.getTableNoCache(viewConn, view1);
             assertEquals(QueryConstants.DIVERGED_VIEW_BASE_COLUMN_COUNT, table.getBaseColumnCount());
             
+            try {
+                viewConn.createStatement().execute("SELECT V2 FROM " + view1);
+                fail("V2 should have been droppped");
+            } catch (SQLException e) {
+                assertEquals(SQLExceptionCode.COLUMN_NOT_FOUND.getErrorCode(), e.getErrorCode());
+            }
+            
             // Add a new regular column and pk column  to the base table
             String alterBaseTable = "ALTER TABLE " + baseTable + " ADD V3 VARCHAR, PK2 VARCHAR PRIMARY KEY";
             conn.createStatement().execute(alterBaseTable);
             
             // Column V3 shouldn't have propagated to the diverged view.
-            String sql = "SELECT V3 FROM " + view1;
             try {
-                viewConn.createStatement().execute(sql);
+                viewConn.createStatement().execute("SELECT V3 FROM " + view1);
```

Comment:
Assert.fail in case this exception is not caught. Similarly for all other places where we expect an exception.